packages/bot-skeleton/src/services/tradeEngine/utils/helpers.js

```diff
@@ -220,3 +282,24 @@ export const createDetails = contract => {
 };
 
 export const getUUID = () => `${new Date().getTime() * Math.random()}`;
+
+const hasBlockOfType = (targetType, workspace) => {
+    const allBlocks = workspace.getAllBlocks();
+    return allBlocks.some(block => block.type === targetType && !!block.parentBlock_);
+};
+
+export const checkBlocksExistenceToSubscribeProposal = () => {
```

Comment:
We can make this more dynamic so in the future we can reuse it to check other blocks. Preferably allow it to accept single or array of block names

Reply:
@prince-deriv to check blocks we have another function` hasBlockOfType()`, we use it like an inner function, but the current method `checkBlocksExistenceToSubscribeProposal()` is needed to store all the results needed for the query proposal in one place. Please note, the point is a little different

Reply:
✅ I've improved the function as we agreed in private messages